File: two_sum/two_sum.php
<?php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
    $seen = [];
    foreach($nums as $index => $value){
        $remaining = $target - $value;
        if(array_key_exists($remaining, $seen)){
            return array($index, $seen[$remaining]);
        }
        $seen[$value] = $index;
    }}
}
$sol = new Solution;
print_r($sol->twoSum([2,7,11,15], 9));
// echo 'Test';

File: two_sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for index, value in enumerate(nums):
            remaining = target - value
           
            if remaining in seen:
                return (index, seen[remaining])
            
            seen[value] = index 